File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hs = set(nums)
        maxCnt = 0
        for i in nums:
            num = i
            cnt = 0
            while num-1 in hs:
                num -= 1
            while num in hs:
                maxCnt = max(maxCnt,cnt + 1)
                hs.remove(num)
                num = num + 1     
                cnt += 1
        return maxCnt

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> hashSet = new HashSet<>();
        for(int num:nums){
            hashSet.add(num);
        }
        int maxLen = 0;
        for(int num: nums){
            if(!hashSet.contains(num-1)){
                int cnt = 1;
                while(hashSet.contains(num+1)){
                    cnt += 1;
                    num += 1;
                }
                maxLen = Math.max(maxLen,cnt);
            }
        }
        return maxLen;
    }
}